### ISSUE TITLE
[RyuJIT/ARM32] Can't run corefx tests with COMPlus_AltJit

### ISSUE THREAD

**Issue by hqueue** [dotnet/coreclr#13326]
When running corefx tests with `COMPlus_AltJit` to test `ryujit/arm32`, test can't start as follows on Rpi3.

```
hqueue@rpi3:/corefx_tests/AnyOS.AnyCPU.Release/System.Buffers.Tests/netstandard$ COMPlus_AltJit=* /coreoverlay/corerun xunit.console.netcore.exe System.Buffers.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Buffers.Tests
System.NullReferenceException: Object reference not set to an instance of an object.
```

Without `COMPlus_AltJit`, it works well.
```
hqueue@rpi3:/corefx_tests/AnyOS.AnyCPU.Release/System.Buffers.Tests/netstandard$ /coreoverlay/corerun xunit.console.netcore.exe System.Buffers.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Buffers.Tests
Discovered:  System.Buffers.Tests
Starting:    System.Buffers.Tests
Finished:    System.Buffers.Tests

=== TEST EXECUTION SUMMARY ===
   System.Buffers.Tests  Total: 77, Errors: 0, Failed: 0, Skipped: 0, Time: 18.411s
```

Does any body know how to run corefx tests with `COMPlus_AltJit` option ?
Note: I've used `corerun` instead of `dotnet` which is supposed to be used for corefx tests as default.

**Comment by hqueue** [comment]
/cc: @BruceForstall Please take a look.  Is it possible to run corefx tests with AltJit ?

**Comment by BruceForstall** [comment]
I haven't ever tried running corefx tests with altjit. Have you tried debugging to see where the failure is? I presume protojit.dll (or libprotojit.so?) exists in the same location as the regular clrjit?

A somewhat unsatisfying option is to (temporarily) change clrjit to be RyuJIT, by changing that build to not define `LEGACY_BACKEND`. You wouldn't get any NYI fallback if you do that.

**Comment by hqueue** [comment]
>  Have you tried debugging to see where the failure is? 

I tried to debug it with `gdb` but got no meaningful clue untiil now.

>  I presume protojit.dll (or libprotojit.so?) exists in the same location as the regular clrjit?

Yes. There is  `libprotojit.so` and I have ran coreclr tests with same coreoverlay using altjit option

> A somewhat unsatisfying option is to (temporarily) change clrjit to be RyuJIT, by changing that build to not define LEGACY_BACKEND. You wouldn't get any NYI fallback if you do that.

I also think about this option but as the  last choice.

**Comment by hqueue** [comment]
@BruceForstall Found a reason. It seems that recent regression in ryujit/arm32 caused the failure.

I'm going to close this after the regression is fixed.

**Comment by hqueue** [comment]
Fixed by #13364
